OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -41,7 +41,7 @@ public void Killed(Actor self, AttackInfo e)
 			if (self.World.SharedRandom.Next(100) > explodesInfo.Chance)
 				return;
 
-			if (explodesInfo.DeathType != null && e.Warhead != null && !explodesInfo.DeathType.Contains(e.Warhead.DeathType))
+			if (explodesInfo.DeathType != null && e.Warhead != null && !explodesInfo.DeathType.Any(x => e.Warhead.DamageTypes.Contains(x)))
```

Comment:
This can be simplified to `!explodesInfo.DeathType.Intersect(e.Warhead.DamageTypes).Any()`